util/java/libtiled-java/src/main/java/org/mapeditor/io/TMXMapReader.java

```diff
@@ -260,14 +271,15 @@ private TileSet unmarshalTileset(Node t, boolean isExternalTileset) throws Excep
         if (source != null) {
             source = replacePathSeparator(source);
             URL url = URLHelper.resolve(xmlPath, source);
-            InputStream in = url.openStream();
-            TileSet ext = unmarshalTilesetFile(in, url);
+            try (InputStream in = StreamHelper.openStream(url)) {
```

Comment:
This seems to be the first time we rely on try-with-resources, introduced in Java 7. Does the library already specify at least Java 7 is required somewhere?

Reply:
Yes, at `org.mapeditor:tiled-parent` pom.xml:
```xml
<maven.compiler.source>1.8</maven.compiler.source>
```
However, I can reorganize the code to properly close the streams without relying on try-with-resources if you prefer.

Reply:
No, I much prefer the new feature, thanks! :-)